examples/demo-rollup/src/test_rpc.rs

```diff
@@ -0,0 +1,205 @@
+use std::net::SocketAddr;
+
+use curl::easy::{Easy2, Handler, List, WriteError};
+use demo_stf::app::{DemoBatchReceipt, DemoTxReceipt};
+use serde::{Serialize, Deserialize};
+use sov_db::ledger_db::{LedgerDB, SlotCommit};
+use sov_rollup_interface::{traits::{CanonicalHash, BlockHeaderTrait}, services::da::SlotData, stf::{BatchReceipt, TransactionReceipt, Event, EventKey}};
+use tokio::{sync::oneshot};
+
+use crate::{config::RpcConfig, ledger_rpc};
+
+struct Collector(Vec<u8>);
+
+impl Handler for Collector {
+    fn write(&mut self, data: &[u8]) -> Result<usize, WriteError> {
+        self.0.extend_from_slice(data);
+        Ok(data.len())
+    }
+}
+
+fn query_test_helper(data: &[u8], expected: &str) {
+    let mut headers = List::new();
+    headers.append("Content-Type: application/json").unwrap();
+
+    let mut easy = Easy2::new(Collector(Vec::new()));
+    easy.http_headers(headers).unwrap();
+    easy.post_fields_copy(data).unwrap();
+    easy.post(true).unwrap();
+
+    easy.url("http://127.0.0.1:12345").unwrap();
+    easy.perform().unwrap();
+
+    assert_eq!(easy.response_code().unwrap(), 200);
+    let contents = easy.get_ref();
+    assert_eq!(String::from_utf8_lossy(&contents.0), expected);
+}
+
+#[derive(Serialize, Deserialize, PartialEq, core::fmt::Debug, Clone)]
+struct TestBlockHeader {
+    prev_hash: [u8; 32],
+}
```

Comment:
This looks like it might be useful for other crates as well. Should we move it into the `rollup-interface` crate under the `mocks` feature?

Reply:
Sure, doing it atm